### PR TITLE
[Feature-14802][api] Use Casdoor SSO to log in, add admin user configuration

### LINKED ISSUE #14802: [Feature][API] Add admin user configuration when Casdoor SSO is added
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Add the admin user configuration when Casdoor SSO is specified, similar to the ldap.user.admin configuration parameter when using ldap login

### Use case

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by gaopeng666**
* Use Casdoor SSO to log in, add admin user configuration

This closes #14802

## Verify this pull request

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

### CODE REVIEW

**Review by EricGao888 — DISMISSED** [review]

**Comment by qingwli** [comment]
Hi @gaopeng666 Could you update the doc about this feature

**Review by SbloodyS — COMMENTED** [review]
Please check the failed UT. @gaopeng666

**Comment by gaopeng666** [comment]
> Please check the failed UT. @gaopeng666

I don't understand the issue with this unit test

**Comment by gaopeng666** [comment]
> Hi @gaopeng666 Could you update the doc about this feature

ok，i did

> Hi @gaopeng666 Could you update the doc about this feature

ok i'm done

**Comment by gaopeng666** [comment]
> Hi @gaopeng666 Could you update the doc about this feature

ok i'm done

**Review by qingwli — COMMENTED** [review]
`architecture/configuration.md`
Please update this doc, thanks

File: dolphinscheduler-api/src/main/resources/application.yaml [hunk 1]
```diff
@@ -204,6 +207,20 @@ security:
...
+  client-id: ""
+  client-secret: ""
+  # The certificate may be multi-line, you can use `|-` for ease
+  certificate: ""
+  # Your organization name added in Casdoor
+  organization-name: ""
+  # Your application name added in Casdoor
+  application-name: ""
+  # Doplhinscheduler login url
+  redirect-url: ""
```
Comment (qingwli) [inline dolphinscheduler-api/src/main/resources/application.yaml:222]: I see the Java code change for casdoor user admin,and how about this config used?

Comment (gaopeng666) [reply dolphinscheduler-api/src/main/resources/application.yaml:222]: Similar to LDAP, determine the user type when creating a local user

Comment (qingwli) [reply dolphinscheduler-api/src/main/resources/application.yaml:222]: It's configuration about casdoor and we don't need to write code for this part?

Comment (gaopeng666) [reply dolphinscheduler-api/src/main/resources/application.yaml:222]: The casdoor function has been submitted before, but the configuration was not submitted. I just added the casdoor user admin parameter and submitted the configuration by the way.

**Comment by qingwli** [comment]
Please fix this ut 
```
2023-09-01T09:25:29.0405973Z [INFO] 
2023-09-01T09:25:29.0406671Z [INFO] Results:
2023-09-01T09:25:29.0407539Z [INFO] 
2023-09-01T09:25:29.0408408Z [ERROR] Errors: 
2023-09-01T09:25:29.0409374Z [ERROR]   CasdoorAuthenticatorTest.testAuthenticate:142 » NullPointer
2023-09-01T09:25:29.0409931Z [INFO] 
2023-09-01T09:25:29.0410403Z [ERROR] Tests run: 622, Failures: 0, Errors: 1, Skipped: 11
2023-09-01T09:25:29.0410865Z [INFO] 
2023-09-01T09:25:29.0427280Z [INFO] ------------------------------------------------------------------------
2023-09-01T09:25:29.0428684Z [INFO] Reactor Summary for dolphinscheduler dev-SNAPSHOT:

**Review by qingwli — APPROVED** [review]
LGTM

**Comment by qingwli** [comment]
Thanks for your contribution @gaopeng666

**Pull request merged**